modules/MiniFigDirectionDetector.h

```diff
@@ -0,0 +1,57 @@
+/**
+ * @file   MinifigDirectionDetector.h
+ * @brief  ミニフィグの向きを検出するクラス
+ * @author nishijima515
+ */
+
+#ifndef MINIFIG_DIRECTION_DETECTOR_H
+#define MINIFIG_DIRECTION_DETECTOR_H
+
+#include <opencv2/opencv.hpp>
+#include <opencv2/dnn.hpp>
+#include <algorithm>
+#include <iostream>
+#include <vector>
+#include <string>
+
+#define CONFIDENCE_THRESHOLD 0.4  // 検出結果を採用する最低信頼度の閾値
+#define NMS_THRESHOLD 0.4         // 検出ボックス同士の重なりを判断する閾値
+
+using namespace cv;
+using namespace dnn;
+
+// ミニフィグの向きを表す列挙体
+enum class MiniFigDirection { FRONT, LEFT, BACK, RIGHT };
+
+struct MiniFigDirectionResult {
+  bool wasDetected = false;    // 検出が成功したかどうか
+  MiniFigDirection direction;  // ミニフィグの向きを表す列挙体
+};
+
+class MiniFigDirectionDetector {
+ public:
+  // コンストラクタ
+  MiniFigDirectionDetector(const std::string& modelPath);
+
+  // ミニフィグの向き検出を行う処理
+  // MiniFigDirectionResult 型の構造体を返す
+  MiniFigDirectionResult detect(const cv::Mat& frame, const std::string& saveImagePath);
```

Comment:
引数で受け取って結果を返す方が柔軟だと思うけど、どう思いますか？
```suggestion
  // MiniFigDirectionResult 型の構造体を返す
  void detect(const cv::Mat& frame, MiniFigDirectionResult& result);
```